src/shared/Task.ts

```diff
@@ -1,22 +1,22 @@
-import { Allow, Entity, Fields, Validators } from "remult";
+import { Allow, describeClass, Entity, Fields, Validators } from "remult";
 import { Roles } from "./Roles";
 
-@Entity<Task>("tasks", {
-      allowApiRead: Allow.authenticated,
-      allowApiUpdate: Allow.authenticated,
-      allowApiInsert: Roles.admin,
-      allowApiDelete: Roles.admin
-})
 export class Task {
-      @Fields.uuid()
       id!: string;
-
-      @Fields.string({
-         validate: Validators.required,
-         allowApiUpdate: Roles.admin
-      })
       title = '';
-
-      @Fields.boolean()
       completed = false;
 }
+
+describeClass(Task, Entity<Task>("tasks", {
+      allowApiRead: Allow.authenticated,
+      allowApiUpdate: Allow.authenticated,
+      allowApiInsert: Roles.admin,
+      allowApiDelete: Roles.admin
+}), {
+      id: Fields.uuid(),
+      title: Fields.string({
+            validate: Validators.required,
+            allowApiUpdate: Roles.admin
+      }),
+      completed: Fields.boolean()
+})
```

Comment:
This looks equivalent to the decorators version. 2 questions:
1. If I have a typo anywhere, does TS catch that and let me know? If no, then does at least the runtime catch it?
2. Have you looked into the mobx way of decorating classes without decorators (in the ctor)? I'm assuming they carefully evaluated different approaches and that their chosen one was found superior.

Reply:
Thanks - Answers:
1. Yes - typescript will let you know there is an error and search references work.
2. Mobx needed something other than me - All I need is some metadata - mobx needed to alter the class instance itself - that's why they used the "makeAutoObservable" in the constructor and changed the members to properties.